core/src/mast/mod.rs

```diff
@@ -527,6 +598,11 @@ impl DecoratorId {
         }
     }
 
+    /// Creates a new [`DecoratorId`] without checking its validity.
+    pub(crate) fn new_unsafe(value: u32) -> Self {
```

Comment:
The convention for this in Rust is `new_unchecked`, anything with `unsafe` in the name should be an `unsafe fn` IMO.

See things like `NonZeroU32::new_unchecked`, `NonNull::new_unchecked`, etc.